lib/engine/edax.dart

```diff
@@ -12,94 +12,125 @@ import 'package:shared_preferences/shared_preferences.dart';
+        File(evalFilePath).writeAsBytesSync(evalData.buffer.asUint8List());
+      }
     }
-  }
-  if (pref.getString('libedaxPath') == null) {
-    var libedaxName = '';
-    if (Platform.isMacOS) libedaxName = 'libedax.dylib';
-    if (Platform.isWindows) libedaxName = 'libedax-x64.dll';
-    if (Platform.isLinux) libedaxName = 'libedax.so';
-    final libedaxData = await rootBundle.load('assets/libedax/dll/$libedaxName');
-    final libedaxPath = '${docDir.path}/$libedaxName';
-    File(libedaxPath).writeAsBytesSync(libedaxData.buffer.asUint8List());
-    await pref.setString('libedaxPath', libedaxPath);
-  } else {
-    final libedaxPath = pref.getString('libedaxPath');
-    if (!File(libedaxPath).existsSync()) {
-      var libedaxName = '';
-      if (Platform.isMacOS) libedaxName = 'libedax.dylib';
-      if (Platform.isWindows) libedaxName = 'libedax-x64.dll';
-      if (Platform.isLinux) libedaxName = 'libedax.so';
-      final libedaxData = await rootBundle.load('assets/libedax/dll/$libedaxName');
-      File(libedaxPath).writeAsBytesSync(libedaxData.buffer.asUint8List());
+
+    // FIXME: remove this.
+    if (Platform.isWindows || Platform.isLinux) {
+      final libedaxData = await _libedaxAssetData;
+      File(await _libedaxPath).writeAsBytesSync(libedaxData.buffer.asUint8List());
     }
+
+    final edax = LibEdax(await _libedaxPath)
+      ..libedaxInitialize([
+        '',
+        '-eval-file',
+        pref.getString(evalFilePathPrefKey),
+        '-book-file',
+        pref.getString(bookFilePathPrefKey),
+      ])
+      ..edaxInit()
+      ..edaxVersion();
+    return edax.edaxGetBoard().prettyString(TurnColor.black);
   }
-  if (Platform.isMacOS) {
-    await Process.run('xattr', ['-d', 'com.apple.quarantine', pref.getString('libedaxPath')], runInShell: true);
+
+  Future<String> get _libedaxPath async {
+    // See: https://flutter.dev/docs/development/platform-integration/c-interop#compiled-dynamic-library-macos
+    if (Platform.isMacOS) return defaultLibedaxName;
+
+    // FIXME: temporary implement.
+    final docDir = await _docDir;
+    if (Platform.isWindows) return '${docDir.path}/$defaultLibedaxName';
+    if (Platform.isLinux) return '${docDir.path}/$defaultLibedaxName';
+    throw Exception('${Platform.operatingSystem} is not supported');
   }
 
-  final edax = LibEdax(pref.getString('libedaxPath'))
-    ..libedaxInitialize(
-      ['', '-eval-file', pref.getString('evalFilePath'), '-book-file', pref.getString('bookFilePath')],
-    )
-    ..edaxInit()
-    ..edaxVersion();
-  return edax.edaxGetBoard().prettyString(TurnColor.black);
+  Future<ByteData> get _libedaxAssetData async => rootBundle.load('assets/libedax/dll/$defaultLibedaxName');
+
+  Future<ByteData> get _evalAssetData async => rootBundle.load('assets/libedax/data/eval.dat');
+
+  // e.g. Mac: ~/Library/Containers/com.example.pedax/Data/Documents
+  Future<Directory> get _docDir async => getApplicationDocumentsDirectory();
```

Comment:
See: https://pub.dev/documentation/path_provider/latest/path_provider/path_provider-library.html

> Path to a directory where the application may place data that is user-generated, or that cannot otherwise be recreated by your application.

`edax.dat`, `book.dat` should be user-customized